### PR TITLE
Update soroban settings upgrade docs

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: docs/software/soroban-settings.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+---
+title: Soroban Settings
+---
+
+Soroban has a large set of settings stored on ledger than can be modified
```
Comment (anupsdf) [inline docs/software/soroban-settings.md:5]: typo: *that can

File: docs/software/soroban-settings.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+through a vlaidator vote. This documents describes how to propose a new settings
```
Comment (anupsdf) [inline docs/software/soroban-settings.md:6]: typo: validator
nit: This document describes

File: docs/software/soroban-settings.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+## Propose a settings upgrade
+
+This section will describe how to propose a settings upgrade, but take a look at
+the [admin page](admin.md#upgrading-soroban-settings) for more information on
+how the settings upgrade mechanism works internally. **If you are just being asked
+to vote for an upgrade, please move on to the [Examine a proposed upgrade](#examine-a-proposed-upgrade) step for details on how to examine a
+proposed upgrade.**
+
+1. Using the stellar-xdr rust tool and a JSON file as input, run `stellar-xdr encode --type ConfigUpgradeSet JSON_FILE`. [phase1.json](../../soroban-settings/phase1.json) can be used as a template to propose upgrades.
```
Comment (anupsdf) [inline docs/software/soroban-settings.md:18]: Should we add a pre-requisite step on how to install stellar-xdr tool?

File: docs/software/soroban-settings.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - Example output - `AAAAAQAAAAsAAAAM` (note that this string can be much larger depending on the settings you're upgrading)
+
+2. Using the output from the command above, run the `get-settings-upgrade-txs PUBLIC_KEY SEQ_NUM NETWORK_PASSPHRASE --xdr CONFIG_UPGRADE_SET_XDR --signtxs` stellar-core command line util. Note that you will be submitting transactions, so the account for the public key specified must exist and have the funds to pay the transaction fees.
```
Comment (anupsdf) [inline docs/software/soroban-settings.md:21]: Should we drop the 'line util' and just saw command? If that is the norm, then we can spell out 'utility'?

File: docs/software/soroban-settings.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    - Example command - `stellar-core get-settings-upgrade-txs GAUQW73V52I2WLIPKCKYXZBHIYFTECS7UPSG4OSVUHNDXEZJJWFXZG56 73014444032 "Public Global Stellar Network ; September 2015" --xdr AAAAAQAAAAsAAAAM --signtxs`
+    - You will be prompted for the secret key that corresponds to the public key passed in. Provide it to sign the transactions required to set up the upgrade.
+    - The output will be 7 lines in this format - <br>
```
Comment (anupsdf) [inline docs/software/soroban-settings.md:24]: How about showing a sample output before explaining the output here?

File: docs/software/soroban-settings.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    - The output will be 7 lines in this format - <br> 
+        &ensp;1.Base64 upload `TransactionEnvelope` XDR<br>
+        &ensp;2.Hex tx ID for the upload tx<br>
+        &ensp;3.Base 64 create `TransactionEnvelope` XDR<br>
+        &ensp;4.Hex tx ID for the create tx<br>
+        &ensp;5.Base64 invoke `TransactionEnvelope` XDR<br>
+        &ensp;6.Hex tx ID for the invoke tx<br>
+        &ensp;7.Base64 `ConfigUpgradeSetKey` XDR<br>
+3. Submit three transactions to the network to set up the upgrade, replacing the blob placeholders below with the `TransactionEnvelopes` in lines 1, 3, and 5.
+    1. `curl -G 'http://localhost:11626/tx' --data-urlencode 'blob=LINE_1_OUTPUT'`
```
Comment (anupsdf) [inline docs/software/soroban-settings.md:33]: How about using less than/greater than around the placeholder text to be replaced? `blob=<LINE_1_OUTPUT>`

**Review by anupsdf — COMMENTED** [review]
Looks good. Added a few comments.

File: docs/software/soroban-settings.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+5. Now schedule the upgrade on all the required validators using the output from line 7 (the `ConfigUpgradeSetKey`) and an agreed upon time in the future
+    1. `curl -G 'http://localhost:11626/upgrades?mode=set&upgradetime=YYYY-MM-DDTHH:MM:SSZ' --data-urlencode 'configupgradesetkey=LINE_7_OUTPUT'`
+
+
+## Examine a proposed upgrade
+
+You can use the `dumpproposedsettings` http command along with a base64 encoded XDR
+serialized `ConfigUpgradeSetKey` to query a proposed upgrade. Example - `curl -G 'http://localhost:11626/dumpproposedsettings' --data-urlencode 'blob=A6MvjFLujnqaZa5hacafWyYwhpk4cgRpyu0z6ilZ0pm1S7fmjSNnsyjGwGodLGiD8ss8S1AHiOBBb6GQbOeMbw=='`
+
+You can also get the current Soroban settings to compare against by using the following command - `curl -G 'http://localhost:11626/sorobaninfo' --data-urlencode 'format=detailed'`.
```
Comment (anupsdf) [inline docs/software/soroban-settings.md:47]: How about have a section heading here to see the current Soroban settings to make it stand out and also have its own hyperlink?
`## Examine current settings` or current Soroban settings?

File: docs/software/soroban-settings.md (hunk 1 above)
Comment (sisuresh) [reply docs/software/soroban-settings.md]: Fixed

File: docs/software/soroban-settings.md (hunk 2 above)
Comment (sisuresh) [reply docs/software/soroban-settings.md]: Fixed

File: docs/software/soroban-settings.md [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+[commands.md](commands.md) doc for more details on the commands used below.
+
+## Propose a settings upgrade
+
+This section will describe how to propose a settings upgrade, but take a look at
+the [admin page](admin.md#upgrading-soroban-settings) for more information on
+how the settings upgrade mechanism works internally. **If you are just being asked
+to vote for an upgrade, please move on to the [Examine a proposed upgrade](#examine-a-proposed-upgrade) step for details on how to examine a
+proposed upgrade.**
+
```
[the remark this answers is not in the document]
Comment (sisuresh) [reply docs/software/soroban-settings.md:17]: I don't think this is necessary.

File: docs/software/soroban-settings.md (hunk 3 above)
Comment (sisuresh) [reply docs/software/soroban-settings.md:18]: Good point. Updated

File: docs/software/soroban-settings.md (hunk 4 above)
Comment (sisuresh) [reply docs/software/soroban-settings.md]: Updated

File: docs/software/soroban-settings.md (hunk 5 above)
Comment (sisuresh) [reply docs/software/soroban-settings.md]: Done

File: docs/software/soroban-settings.md (hunk 6 above)
Comment (sisuresh) [reply docs/software/soroban-settings.md]: Done

File: docs/software/soroban-settings.md (hunk 7 above)
Comment (sisuresh) [reply docs/software/soroban-settings.md]: Done

**Review by anupsdf — APPROVED** [review]

**Comment by dmkozh** [comment]
r+ 22685cd

**Pull request merged**